numba_cuda/numba/cuda/cudadrv/driver.py

```diff
@@ -2434,39 +2467,51 @@ def launch_kernel(
     args,
     cooperative=False,
 ):
-    param_ptrs = [addressof(arg) for arg in args]
-    params = (c_void_p * len(param_ptrs))(*param_ptrs)
-
-    params_for_launch = addressof(params)
-    extra = 0
-
-    if cooperative:
-        driver.cuLaunchCooperativeKernel(
-            cufunc_handle,
-            gx,
-            gy,
-            gz,
-            bx,
-            by,
-            bz,
-            sharedmem,
-            hstream,
-            params_for_launch,
-        )
+    # Convert stream handle to cuda.core Stream object
+    if hstream is None:
+        hstream = 0
+    elif isinstance(hstream, binding.CUstream):
+        hstream = get_cuda_native_handle(hstream)
+    elif not isinstance(hstream, int):
+        hstream = 0
+    stream = ExperimentalStream.from_handle(hstream)
+
+    # Configure kernel launch parameters
+    config = LaunchConfig(
+        grid=(int(gx), int(gy), int(gz)),
+        block=(int(bx), int(by), int(bz)),
```

Comment:
This would be a @gmarkall question: Is `driver.launch_kernel` a public API? If not, is it possible that we do not unpack grid/block at the call sites? `cuda.core.LaunchConfig` prefers the 3-tuples.

In the meanwhile... 
```suggestion
        grid=(gx, gy, gz),
        block=(bx, by, bz),
```

Reply:
A minor trick is to cache a `LaunchConfig` instance in the dispatcher so that we don't recreate it at every launch.